airbyte-cdk/bulk/toolkits/load-object-storage/src/test/kotlin/io/airbyte/cdk/load/file/object_storage/ObjectStoragePathFactoryUTest.kt

```diff
@@ -4,42 +4,59 @@
 
 package io.airbyte.cdk.load.file.object_storage
 
+import io.airbyte.cdk.load.command.Append
 import io.airbyte.cdk.load.command.DestinationStream
 import io.airbyte.cdk.load.command.object_storage.ObjectStoragePathConfiguration
 import io.airbyte.cdk.load.command.object_storage.ObjectStoragePathConfigurationProvider
+import io.airbyte.cdk.load.data.ObjectTypeWithEmptySchema
 import io.airbyte.cdk.load.file.TimeProvider
 import io.airbyte.cdk.load.file.object_storage.ObjectStoragePathFactory.Companion.DEFAULT_FILE_FORMAT
 import io.airbyte.cdk.load.file.object_storage.ObjectStoragePathFactory.Companion.DEFAULT_PATH_FORMAT
 import io.airbyte.cdk.load.file.object_storage.ObjectStoragePathFactory.Companion.DEFAULT_STAGING_PREFIX_SUFFIX
 import io.airbyte.cdk.load.state.object_storage.ObjectStorageDestinationState.Companion.OPTIONAL_ORDINAL_SUFFIX_PATTERN
-import io.mockk.coEvery
+import io.mockk.every
 import io.mockk.impl.annotations.MockK
 import io.mockk.mockk
 import java.util.stream.Stream
 import org.junit.jupiter.api.Assertions.assertEquals
 import org.junit.jupiter.api.Assertions.assertNotNull
 import org.junit.jupiter.api.Assertions.assertNull
-import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.params.ParameterizedTest
 import org.junit.jupiter.params.provider.Arguments
 import org.junit.jupiter.params.provider.MethodSource
 
 class ObjectStoragePathFactoryUTest {
-    @MockK lateinit var stream: DestinationStream
-    @MockK lateinit var pathConfigProvider: ObjectStoragePathConfigurationProvider
-    @MockK lateinit var timeProvider: TimeProvider
-
-    @BeforeEach
-    fun setup() {
-        coEvery { stream.descriptor } returns DestinationStream.Descriptor("test", "stream")
-        coEvery { timeProvider.syncTimeMillis() } returns 0
-        coEvery { timeProvider.currentTimeMillis() } returns 1
+    private val timeProvider: TimeProvider = MockTimeProvider()
+    private val stream: DestinationStream =
+        DestinationStream(
+            DestinationStream.Descriptor("test", "stream"),
+            Append,
+            ObjectTypeWithEmptySchema,
+            0L,
+            0L,
+            101
+        )
+    @MockK(relaxed = true) lateinit var pathConfigProvider: ObjectStoragePathConfigurationProvider
+
+    // This is flaky if used as a mockk mock.
+    class MockTimeProvider : TimeProvider {
+        override fun syncTimeMillis(): Long {
+            return 0
+        }
+
+        override fun currentTimeMillis(): Long {
+            return 1
+        }
+
+        override suspend fun delay(ms: Long) {
+            TODO("Not yet implemented")
```

Comment:
nit, since we're not planning to actually implement it
```suggestion
            throw NotImplementedError()
```

Reply:
check is taking >40m rn, if it fails I'll make these changes.